src/handlers/window_handler.rs

```diff
@@ -307,3 +317,41 @@ pub fn get_next_or_previous(manager: &mut Manager, handle: &WindowHandle) -> Opt
     }
     None
 }
+
+// Get size and position of scratchpad from config and workspace size
+pub fn scratchpad_xyhw(xyhw: &Xyhw, scratch_pad: &ScratchPad) -> Xyhw {
+    let scratch_pad_x = scratch_pad.x.unwrap_or(25);
+    let scratch_pad_y = scratch_pad.y.unwrap_or(25);
+    let scratch_pad_height = scratch_pad.height.unwrap_or(50);
+    let scratch_pad_width = scratch_pad.width.unwrap_or(50);
+
+    let x_sane = if (10..90).contains(&scratch_pad_x) {
+        scratch_pad_x
+    } else {
+        25
+    };
```

Comment:
Not an actual critique on this PR, but asking to improve my own understanding of code quality/best practices/good style:
Would it be good to extract this to
```rust
fn sane_dimension(config_value: Option<i32>, default_value: i32 {
    match config_value {
       Some(dim) if (10..90).contains(&dim) => dim,
       _ => default_value,
    }
}
```
and then call `let x_sane = sane_dimension(scratch_pad_x, 25)` and so on?
(Writing on mobile, so please excuse if this is actually broken code 🤷🏼‍♂️, not entirely sure about that `Some() if`)

Reply:
Great idea and thanks for your help! I extracted sane_dimensions and now even more cleaner.
Meanwhile, I tried with impl Default for Config before. But the problem is, when no scratchpad is configured, the default will still work and get a default value. Probably it's best to show nothing when no scratchpad is configured.

Reply:
Because I have to face a situation: when people upgrade, they'll not change configuration file unless necessary. Thus current solution can work with the situation when scratchpad is configured, but no xyhw is provided. Here a sane defaults will be provided, same as behavior before the patch.

Reply:
Well I believe this is exactly the purpose of the default config and the reason why I suggested `Option<i32>` so the parser can handle backwards compatibility.
I'm not sure if this wold work but I'd try to put `scratchpad.x: 50` and so on in the `Self {}` element in `impl Default for Config`

Reply:
That's what I thought might work, but now that I rethink it I guess you are right, that this won't work as intended.
My idea was to define the sane defaults globally so when the config for the scratchpad is set each scratchpad can either take the values from the parser or fallback.
Setting this default values would naturally belong in the above section I guess.
Unfortunately I won't make it to may dev machine anymore today, so I just can look at this on mobile.
If you feel you're stuck here I guess leaving a TODO note in the `Default for Config` to refactor this at a later point and where the default is currently applied, might be ok for now.

As far as I read into branch 4 of @cecton's libification rework a few things might get quite a bit clearer along the way.